--- a
+++ b
@@ -4,6 +4,7 @@
 	"time"
 
 	"github.com/fullstack-lang/gong/test/test/go/models"
+	// injection point for ident package import declaration{{ImportPackageDeclaration}}
 )
 
 // generated in order to avoid error in the package import
@@ -9,6 +10,8 @@
 // generated in order to avoid error in the package import
 // if there are no elements in the stage to marshall
 var _ time.Time
+
+// _ point for meta package dummy declaration{{ImportPackageDummyDeclaration}}
 
 // When parsed, those maps will help with the renaming process
 var _ map[string]any = map[string]any{
@@ -18,10 +21,10 @@
 // function will stage objects
 func _(stage *models.Stage) {
 
+	const __write__local_time = "2025-07-30 06:45:16.339151 EDT"
+	const __write__utc_time__ = "2025-07-30 10:45:16.339151 UTC"
-	const __write__local_time = "2025-06-25 05:34:51.790237 CEST"
-	const __write__utc_time__ = "2025-06-25 03:34:51.790237 UTC"
 
+	const __commitId__ = "0000000091"
-	const __commitId__ = "0000000089"
 
 	// Declaration of instances to stage
 
